compiler/rustc_trait_selection/src/solve/eval_ctxt/mod.rs

```diff
@@ -319,6 +331,20 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
         source: GoalSource,
         goal: Goal<'tcx, ty::Predicate<'tcx>>,
     ) -> Result<(bool, Certainty), NoSolution> {
+        let (normalization_nested_goals, has_changed, certainty) =
+            self.evaluate_goal_raw(goal_evaluation_kind, source, goal)?;
+        assert!(normalization_nested_goals.is_empty());
+        Ok((has_changed, certainty))
+    }
+
+    /// FIXME(-Znext-solver=coinduction): `_source` is currently unused but will
+    /// be necessary once we implement the new coinduction approach.
```

Comment:
don't need to be doc, this needs real doc comments tho

```suggestion
    // FIXME(-Znext-solver=coinduction): `_source` is currently unused but will
    // be necessary once we implement the new coinduction approach.
```